### PR TITLE
Naveen Tg - Feed API Challenge

### PR DESCRIPTION
**Pull request by NaveenTg**
Hey @caiozullo and @mapostolakis
Can you guys take a look on my implementation for Feed API Challenged.
Looking forward for feedback!.

Thanks

### CODE REVIEW

File: FeedAPIChallenge/Feed API/RemoteFeedLoader.swift [hunk 1]
```diff
@@ -20,5 +20,43 @@ public final class RemoteFeedLoader: FeedLoader {
...
+	let url: URL
+	
+	private enum CodingKeys:String, CodingKey{
+		case id = "image_id"
+		case description = "image_desc"
+		case location = "image_loc"
+		case url = "image_url"
+	}
+	
+}
```
Comment (mapostolakis) [inline FeedAPIChallenge/Feed API/RemoteFeedLoader.swift:62]: These types have already been defined in `FeedImageMapper.swift`. You can remove them from this file.

File: FeedAPIChallenge/Feed API/FeedImageMapper.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//
+//  FeedImageMapper.swift
+//  FeedAPIChallenge
+//
+//  Created by Navi on 13/02/21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
```
Comment (mapostolakis) [inline FeedAPIChallenge/Feed API/FeedImageMapper.swift:10]: There is an extra line break here.

File: FeedAPIChallenge/Feed API/FeedImageMapper.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+final class FeedImageMapper{
```
Comment (mapostolakis) [inline FeedAPIChallenge/Feed API/FeedImageMapper.swift:12]: The repo code convention dictates leaving a space before a brace, i.e. `FeedImageMapper {`, `items.map {` 

Please update any occurrences in the code to follow the convention.

**Review by mapostolakis — COMMENTED** [review]
Hi @NaveenTg, you almost passed the challenge!

Please, check my comments and let me know once you're done so I can review it again. Thank you 🙂

**Comment by NaveenTg** [comment]
@mapostolakis 
Thanks for your input!

I forgot to remove struct from RemoteFeedLoader😅
and i changed it to repo code convention.

File: FeedAPIChallenge/Feed API/RemoteFeedLoader.swift [hunk 4]
```diff
@@ -20,5 +20,17 @@ public final class RemoteFeedLoader: FeedLoader {
...
+				completion(FeedImageMapper.map(data, from: response))
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		}
+		
 	}
 }
+
+
```
Comment (mapostolakis) [inline FeedAPIChallenge/Feed API/RemoteFeedLoader.swift:36]: There are a couple of extra line breaks.

File: FeedAPIChallenge/Feed API/FeedImageMapper.swift [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+//  Created by Navi on 13/02/21.
+//  Copyright © 2021 Essential Developer Ltd. All rights reserved.
+//
+
+import Foundation
+
+final class FeedImageMapper{
+	
+	private struct Entity: Decodable {
+		let items:[ImageItem]
```
Comment (mapostolakis) [inline FeedAPIChallenge/Feed API/FeedImageMapper.swift:14]: There is a space missing after the colon.

**Review by mapostolakis — COMMENTED** [review]
Hi @NaveenTg,

Congratulations, you've passed the challenge 🥇

All tests are passing, so your implementation complies with the specs.

I left a couple of comments. Feel free to push new changes.

Great job! ✅

**Pull request closed without merging**